app/controllers/administrative-units/new.js

```diff
@@ -0,0 +1,122 @@
+
+  @dropTask
+  *createAdministrativeUnitTask(event) {
+    event.preventDefault();
+
+    let {
+      administrativeUnit,
+      centralWorshipService,
+      worshipService,
+      primarySite,
+      address,
+      contact,
+      identifierSharepoint,
+      identifierKBO,
+      structuredIdentifierSharepoint,
+      structuredIdentifierKBO,
+    } = this.model;
+
+    yield structuredIdentifierSharepoint.save();
+    yield structuredIdentifierKBO.save();
+
+    identifierSharepoint.structuredIdentifier = structuredIdentifierSharepoint;
+    identifierSharepoint.idName = IDNAMES.SHAREPOINT;
+    yield identifierSharepoint.save();
+
+    identifierKBO.structuredIdentifier = structuredIdentifierKBO;
+    identifierKBO.idName = IDNAMES.KBO;
+    yield identifierKBO.save();
+
+    yield contact.save();
+
+    address.fullAddress = combineFullAddress(address);
+    yield address.save();
+
+    primarySite.address = address;
+    primarySite.contacts.pushObjects([contact]);
+    yield primarySite.save();
+
+    administrativeUnit.identifiers.pushObjects([
+      identifierKBO,
+      identifierSharepoint,
+    ]);
+    administrativeUnit.primarySite = primarySite;
+
+    if (
+      administrativeUnit.classification.get('id') ==
+      CLASSIFICATION.CENTRAL_WORSHIP_SERVICE
+    ) {
+      centralWorshipService.name = administrativeUnit.name;
+      centralWorshipService.recognizedWorshipType =
+        administrativeUnit.recognizedWorshipType;
+      centralWorshipService.classification = administrativeUnit.classification;
+      centralWorshipService.organizationStatus =
+        administrativeUnit.organizationStatus;
+      centralWorshipService.scope = administrativeUnit.scope;
+      centralWorshipService.isSubOrganizationOf =
+        administrativeUnit.isSubOrganizationOf;
+      centralWorshipService.isAssociatedWith =
+        administrativeUnit.isAssociatedWith;
+      centralWorshipService.identifiers = administrativeUnit.identifiers;
+      centralWorshipService.primarySite = administrativeUnit.primarySite;
+
+      yield centralWorshipService.save();
+    } else if (
+      administrativeUnit.classification.get('id') ==
+      CLASSIFICATION.WORSHIP_SERVICE
+    ) {
+      worshipService.name = administrativeUnit.name;
+      worshipService.recognizedWorshipType =
+        administrativeUnit.recognizedWorshipType;
+      worshipService.classification = administrativeUnit.classification;
+      worshipService.organizationStatus = administrativeUnit.organizationStatus;
+      worshipService.scope = administrativeUnit.scope;
+      worshipService.isSubOrganizationOf =
+        administrativeUnit.isSubOrganizationOf;
+      worshipService.isAssociatedWith = administrativeUnit.isAssociatedWith;
+      worshipService.identifiers = administrativeUnit.identifiers;
+      worshipService.primarySite = administrativeUnit.primarySite;
+
+      yield worshipService.save();
+    }
+
+    this.router.transitionTo('administrative-units');
```

Comment:
I would redirect to the details page of the newly created administrative unit. We should be able to access the id after the record is saved.